Kernel/Process.cpp

```diff
@@ -790,9 +790,20 @@ void Process::terminate_due_to_signal(u8 signal)
 KResult Process::send_signal(u8 signal, Process* sender)
 {
     InterruptDisabler disabler;
-    // FIXME: PID/TID BUG
-    if (auto* thread = Thread::from_tid(m_pid.value())) {
-        thread->send_signal(signal, sender);
+    Thread* receiver_thread;
+    // Try to send it to the "obvious" main thread:
+    receiver_thread = Thread::from_tid(m_pid.value());
+    // If the main thread has died, there may still be other threads:
+    if (!receiver_thread) {
+        // The first one should be good enough.
+        // Neither kill(2) nor kill(3) specify any selection precedure.
```

Comment:
I think we should also check that the receiving thread doesn't block this signal.

Reply:
I believe `Thread::send_signal` handles that? If not, the bug was already there before the commit.

Also, now that I think more deeply about that, this introduces non-determinism: Assume a process with three threads: Main, T1, and T2. Main does `pthread_exit`. T1 (but not T2) masks a signal, say SIGINT. Then the process receives a SIGINT. Now it's implementation-defined which thread receives the signal, and thus whether the process dies or not. (Currently it will be received in the chronologically-first thread, but only due to the iteration order of `Process::for_each_thread`.)

Maybe POSIX is right and we should do it differently:

> ["The signal disposition is a per-process attribute: in a multithreaded application, the disposition of a particular signal is the same for all threads."](https://linux.die.net/man/7/signal)